action.yaml

```diff
@@ -20,5 +20,37 @@ outputs:
     description:
       The version of Terraform that has been installed and is ready for use.
 runs:
-  using: "node20"
-  main: "dist/main/index.js"
+  using: "composite"
+  steps:
+    - name: Read node version
+      id: get-node-version
+      run: |
+        # Check if an .nvmrc file exists and export the node version if it does
+        if [ -f .nvmrc ]; then
+          NODE_VERSION=$(cat .nvmrc)
+          echo "Node version from .nvmrc: $NODE_VERSION"
+          echo "node_version=$NODE_VERSION" >> $GITHUB_OUTPUT
+        fi
+
+        # Check if a .node-version file exists and export the node version if it does
+        if [ -f .node-version ]; then
+          NODE_VERSION=$(cat .node-version)
+          echo "Node version from .node-version: $NODE_VERSION"
+          echo "node_version=$NODE_VERSION" >> $GITHUB_OUTPUT
+        fi
+      shell: bash
+    - uses: actions/setup-node@v4
```

Comment:
With this action we get cached versions out of the box and a more reliable infrastructure that doesn't require us to rebuild runners to cache new node versions: https://github.com/actions/setup-node